src/interfaces/token.ts

```diff
@@ -240,28 +240,46 @@ export class TokenInterface<C extends ChainId> extends ServiceInterface<C> {
   }
 
   /**
-   * Approve the token amount that spender is allowed to spend on behalf of owner
+   * Returns the populated transaction to approve the token amount that spender is allowed to spend on behalf of owner
    * @param ownerAddress
    * @param tokenAddress
    * @param spenderAddress
    * @param amount
    * @param overrides
-   * @returns TokenAllowance
+   * @returns PopulatedTransaction
    */
-  async approve(
+  async populateApprove(
     ownerAddress: Address,
     tokenAddress: Address,
     spenderAddress: Address,
     amount: Integer,
     overrides: CallOverrides = {}
-  ): Promise<TransactionResponse> {
+  ): Promise<PopulatedTransaction> {
     if (isNativeToken(tokenAddress)) throw new SdkError(`Native tokens cant be approved: ${tokenAddress}`);
     if (tokenAddress === spenderAddress) throw new SdkError(`Cant approve token as its spender: ${tokenAddress}`);
 
     const signer = this.ctx.provider.write.getSigner(ownerAddress);
     const tokenContract = new Contract(tokenAddress, TokenAbi, signer);
-    const tx = await tokenContract.populateTransaction.approve(spenderAddress, amount, overrides);
+    return await tokenContract.populateTransaction.approve(spenderAddress, amount, overrides);
```

Comment:
I think `await` is unnecessary since we are returning a promise 
```suggestion
    return tokenContract.populateTransaction.approve(spenderAddress, amount, overrides);
```

Reply:
removed here https://github.com/yearn/yearn-sdk/pull/283/commits/9bf9ba92bc6117e069cae94bc678a1cb43078b8c